packages/camera/camera/example/ios/RunnerTests/CameraDispatchQueueRaceConditionTests.m

```diff
@@ -0,0 +1,41 @@
+//
+//  CameraDispatchQueueRaceConditionTests.m
+//  RunnerTests
+//
+//  Created by Huan Lin on 12/16/21.
+//  Copyright © 2021 The Flutter Authors. All rights reserved.
+//
+
+@import camera;
+@import camera.Test;
+@import XCTest;
+
+@interface CameraDispatchQueueRaceConditionTests : XCTestCase
+@end
+
+@implementation CameraDispatchQueueRaceConditionTests
+
+- (void)testRaceCondition {
+  CameraPlugin *camera = [[CameraPlugin alloc] initWithRegistry:nil messenger:nil];
+
+  XCTestExpectation *disposeExpectation =
+      [[XCTestExpectation alloc] initWithDescription:@"dispose's result must be called"];
+  FlutterMethodCall *disposeCall = [FlutterMethodCall methodCallWithMethodName:@"dispose"
+                                                                     arguments:nil];
+  FlutterMethodCall *createCall = [FlutterMethodCall
+      methodCallWithMethodName:@"create"
+                     arguments:@{@"resolutionPreset" : @"medium", @"enableAudio" : @(1)}];
+  // Mimic a dispose call followed by a create call, which can be triggered by slightly dragging the
+  // home bar, causing the app to be inactive, and immediately regain active.
+  [camera handleMethodCall:disposeCall
+                    result:^(id _Nullable result) {
+                      [disposeExpectation fulfill];
+                    }];
+  [camera handleMethodCall:createCall
+                    result:^(id _Nullable result){
+                    }];
+  [self waitForExpectations:@[ disposeExpectation ] timeout:1];
+  XCTAssertNotNil(camera.dispatchQueue, @"dispatchQueue must not be nil after create method. ");
+}
+
+@end
```

Comment:
You've added a new test file, but not added it to the project, so it's not actually going to be run.